File: controller/Persona.php
<?php

namespace controller;

use libs\Request;
use libs\DB;

class Persona {

  /**
   * Muestra un listado del recurso.
   * @param  \libs\Request  $request
   */
  public function index(Request $request) {
    // Abrimos la base de datos
    $db = DB::connect(/*db*/);

    $search = '%' . trim($request->search) . '%';
    $sql = '
        select * from persona 
        where concat(nombre, apellidos) like ? 
        order by nombre asc'
    ;

    // Busca registros con el criterio de busqueda 'search'.
    $cursor = $db->query( $sql, [$search] );
    return $cursor->fetchAll();
  }

  /**
   * Muestra un recurso por su identificador.
   * @param  \libs\Request  $request
   */
  public function find(Request $request) {
    // Abrimos la base de datos
    $db = DB::connect(/*db2*/);

    $cursor = $db->cursor('select * from persona where id = ?');
    $cursor->execute([ $request->id ]);

    $record = $cursor->fetchOne();
      // Muestra el json.
    return $this->response(true, $record ?$record :null, 'modelo persona');
  }

  /**
   * Inserta el recurso.
   * @param  \libs\Request  $request
   */
  public function insert(Request $request) {
    // Abrimos la base de datos
    $db = DB::connect();
/*
    $cursor = $db->cursor('
        insert into persona(nombre, apellidos)
        values(?, ?)
    ');

    $success = $cursor->execute([
      	$request->nombre,
      	$request->apellidos
    ]);
*/

/*
    $json = json_decode($request->rawPostData());
    $id = $db->insert('persona', $json);
    $success = $id > 0;
*/

    $id = $db->insert('persona', [
      'nombre'    => $request->nombre,
      'apellidos' => $request->apellidos
    ]);

    $success = $id > 0;

    // Muestra el json.
    return $this->response($success, $id);
  }

  /**
   * Modifica el recurso.
   * @param  \libs\Request  $request
   */
  public function update(Request $request) {
    // Abrimos la base de datos
    $db = DB::connect();
/*
    $cursor = $db->cursor('
      	update persona set
      	  nombre = ?,
     	    apellidos = ?
      	where id = ?
    ');

    $success = $cursor->execute([
      	$request->nombre,
      	$request->apellidos,
      	$request->id
    ]);
*/

/*
    $json = json_decode($request->rawPostData());
    $rows = $db->update('persona', $json, 'id = ' . $json->id);
    $success = $rows > 0;
*/

    $rows = $db->update('persona', [
      'nombre'    => $request->nombre,
      'apellidos' => $request->apellidos
    ],'id = ' . $request->id);
    
    $success = $rows > 0;

    // Muestra el json.
    return $this->response($success, $rows);
  }

  /**
   * Elimina el recurso por su identificador.
   * @param  \libs\Request  $request
   */
  public function delete(Request $request) {
    // Abrimos la base de datos
    $db = DB::connect();
/*
    $cursor = $db->cursor('delete from persona where id = ?');
    $success = $cursor->execute([ $request->id ]);
*/
    $rows = $db->delete('persona', 'id = ' . $request->id);
    $success = $rows > 0;
    
    // Muestra el json.
    return $this->response($success, $rows);
  }


  public function response($success, $result = NULL, $message = NULL) {
    return [
      'success' => $success,
      'result' => $result,
      'message' => $message
    ];
  }
}

File: client/java/restlight/WebServiceClient/src/Persona.java

import com.google.gson.annotations.SerializedName;
import com.google.gson.reflect.TypeToken;
import restlight.Call;
import restlight.FormBody;
import restlight.Request;

public class Persona 
{
  @SerializedName("id")
  public long id;
  
  @SerializedName("nombre")
  public String nombre;
  
  @SerializedName("apellidos")
  public String apellidos;

  @Override public String toString() {
    return "Model{" + "id=" + id + ", nombre=" + nombre + ", apellidos=" + apellidos + '}';
  }

  public static class Dao 
  {
    /**
     * Obtiene todo los registros.
     *
     * @return Call
     */
    public Call<Persona[]> search(String search) {
      FormBody body = new FormBody()
            .add("search", search); // ?search={search}
      
      Request request = new Request(
              "GET", "http://127.0.0.1/WebServiceDemo/persona/index", body);
      
      return WebService.newCall(request, Persona[].class);
    }

    /**
     * Obtiene un registro por su id.
     *
     * @param id
     * @return Call
     */
    public Call<Result<Persona>> findById(long id) {
      FormBody body = new FormBody()
            .add("id", id); // ?id={id}
        
      Request request = new Request(
              "GET", "http://127.0.0.1/WebServiceDemo/persona/find", body);
      
      return WebService.newCall(request, new TypeToken< Result<Persona> >(){});
    }

    /**
     * Inserta un registro.
     *
     * @param nombre
     * @param apellidos
     * @return Call
     */
    public Call<Result<Integer>> insert(String nombre, String apellidos) {
      FormBody body = new FormBody()
            .add("nombre", nombre)
            .add("apellidos", apellidos);
        
      Request request = new Request(
              "POST", "http://127.0.0.1/WebServiceDemo/persona/insert", body);

      return WebService.newCall(request, new TypeToken< Result<Integer> >(){});
    }

    /**
     * Modifica un registro.
     *
     * @param nombre
     * @param apellidos
     * @return Call
     */
    public Call<Result<Integer>> update(int id, String nombre, String apellidos) {
      FormBody body = new FormBody()
            .add("id", id)
            .add("nombre", nombre)
            .add("apellidos", apellidos);
        
      Request request = new Request(
              "PUT", "http://127.0.0.1/WebServiceDemo/persona/update", body);

      return WebService.newCall(request, new TypeToken< Result<Integer> >(){});
    }

    /**
     * Elimina un registro
     *
     * @param id
     * @return Call
     */
    public Call<Result<Integer>> delete(long id) {
      FormBody body = new FormBody()
            .add("id", id); // ?id={id}
      
      Request request = new Request(
              "DELETE", "http://127.0.0.1/WebServiceDemo/persona/delete", body);
      
      return WebService.newCall(request, new TypeToken< Result<Integer> >(){});
    }

    
  }
}
